internal/uvm/types.go

```diff
@@ -80,9 +74,12 @@ type UtilityVM struct {
 
 	// VPMEM devices that are mapped into a Linux UVM. These are used for read-only layers, or for
 	// booting from VHD.
-	vpmemDevices      [MaxVPMEMCount]*vpmemInfo // Limited by ACPI size.
-	vpmemMaxCount     uint32                    // The max number of VPMem devices.
-	vpmemMaxSizeBytes uint64                    // The max size of the layer in bytes per vPMem device.
+	vpmemMaxCount     uint32 // The max number of VPMem devices.
+	vpmemMaxSizeBytes uint64 // The max size of the layer in bytes per vPMem device.
+	vpmemOptimized    bool
+	// TODO: check if we can wrap with an interface
```

Comment:
What's the status here?

Reply:
I'll remove the TODO, since I think we might end up in a similar rabbit hole as we did when I initially implemented a `controller` interface... it's probably fine as it is right now. let me know what you think